spec/models/recoverable_version_spec.rb

```diff
@@ -0,0 +1,120 @@
+require "spec_helper"
+
+describe RecoverableVersion do
+
+  describe ".find" do
+    it "returns a RecoverableVersion that wraps the PaperTrail::Version with the given ID" do
+      deleted_evidence = create(:evidence)
+      deleted_evidence.destroy
+      destroy_revision = deleted_evidence.versions.last
+      r_version = RecoverableVersion.find(destroy_revision)
+      expect(r_version).to be_a RecoverableVersion
+      expect(r_version.version).to eq destroy_revision
+    end
+
+    it "returns an error if the version type is not 'destroy'" do
+      evidence        = create(:evidence)
+      create_revision = evidence.versions.first
+      expect do
+        RecoverableVersion.find(create_revision.id)
+      end.to raise_error ActiveRecord::RecordNotFound
+    end
+  end
+
+  describe "#recover" do
+    it "recovers the deleted item and returns true" do
+      deleted_evidence = create(:evidence)
+      deleted_evidence.destroy
+      version = RecoverableVersion.new(deleted_evidence.versions.last)
+      expect(Evidence.exists?(deleted_evidence.id)).to be false
+      expect(version.recover).to be true
+      expect(Evidence.exists?(deleted_evidence.id)).to be true
+      recovered_evidence = version.object
+      expect(recovered_evidence.content).to eq deleted_evidence.content
+    end
+
+    describe "recovering an Evidence whose Issue has been deleted" do
+      it "recovers the Issue as well" do
+        evidence = create(:evidence)
+        issue    = evidence.issue
+
+        evidence.destroy
+        issue.destroy
+
+        version = RecoverableVersion.new(evidence.versions.last)
+        expect(Evidence.exists?(evidence.id)).to be false
+        expect(Issue.exists?(issue.id)).to be false
+        expect(version.recover).to be true
+        expect(Evidence.exists?(evidence.id)).to be true
+        expect(Issue.exists?(issue.id)).to be true
+
+        recovered_evidence = version.object
+        expect(recovered_evidence.issue).to eq issue
+      end
+    end
+  end
+
+  describe ".all" do
+    before do
+      @evidence = create(:evidence)
+      @note     = create(:note)
+      @issue    = create(:issue)
+
+      @deleted_evidence = create(:evidence)
+      @deleted_note     = create(:note)
+      @deleted_issue    = create(:issue)
+
+      # More issue/note hackery :(
```

Comment:
Another proposal: whenever we have to do something like this, mark it with a unique comment (e.g. `FIXME - ISSUE/NOTE INHERITANCE`) - and add the same comment to any existing places with Issue/Note hackery, as we remember them. That way when we get around to sorting out the Issue/Note hierarchy issue, we can easily find all the places like this that will need updating.